AWSPinpoint/AWSPinpointTargetingClient.m

```diff
@@ -243,8 +323,14 @@ - (void)addMetric:(NSNumber *)theValue
     
     @synchronized(self) {
         [self.globalMetrics setValue:theValue forKey:theKey];
-        [self.context.configuration.userDefaults setObject:self.globalMetrics forKey:AWSPinpointEndpointMetricsKey];
-        [self.context.configuration.userDefaults synchronize];
+        NSError *codingError;
+        NSData *globalMetricsData = [AWSNSCodingUtilities versionSafeArchivedDataWithRootObject:self.globalMetrics
+                                                                            requiringSecureCoding:YES
+                                                                                            error:&codingError];
+        if (codingError) {
+            AWSDDLogError(@"Error archiving globalMetricsData. Updating service but not persisting locally: %@", codingError);
```

Comment:
This is still persisting even though there's an error

Reply:
Missed it, should be fixed now.